tests/unit/onyxMultiMergeWebStorageTest.js

```diff
@@ -0,0 +1,70 @@
+import OnyxCache from '../../lib/OnyxCache';
+import waitForPromisesToResolve from '../utils/waitForPromisesToResolve';
+import localforageMock from '../../__mocks__/localforage';
+
+const ONYX_KEYS = {
+    COLLECTION: {
+        TEST_KEY: 'test_',
+    },
+};
+
+const initialTestObject = {a: 'a'};
+const initialData = {
+    test_1: initialTestObject,
+    test_2: initialTestObject,
+    test_3: initialTestObject,
+};
+
+localforageMock.setInitialMockData(initialData);
+
+describe('Onyx.mergeCollection()', () => {
+    let Onyx;
+    let Storage;
+    beforeAll(() => {
+        jest.mock('../../lib/storage');
+        Onyx = require('../../index').default;
+        Storage = require('../../lib/storage').default;
+        jest.useRealTimers();
+    });
+
+    it('mergeCollection', () => {
+        expect(Storage.name).toBe('WebStorage');
+
+        Onyx.init({
+            keys: ONYX_KEYS,
+            registerStorageEventListener: () => {},
+            initialKeyStates: {},
+        });
+
+        const additionalDataOne = {b: 'b', c: 'c'};
+        Onyx.mergeCollection(ONYX_KEYS.COLLECTION.TEST_KEY, {
+            test_1: additionalDataOne,
+            test_2: additionalDataOne,
+            test_3: additionalDataOne,
+        });
+
+        const additionalDataTwo = {d: 'd'};
+        Onyx.mergeCollection(ONYX_KEYS.COLLECTION.TEST_KEY, {
+            test_1: additionalDataTwo,
+            test_2: additionalDataTwo,
+            test_3: additionalDataTwo,
+        });
```

Comment:
I'm not sure this captures the bug with cache
If you revert `mergeCollection` changes would the test fail because `finalObject` is not in OnyxCache? 

I think to cover the cache bug the test should:
1. merge `additionalDataOne` 
2. wait for the merge to complete
3. evict test_1, test_2 from cache
4. merge `additionalDataTwo`
5. Assert `finalObject` is in cache

Reply:
That sounds like a good additional test. Just to clarify this is covering the case where we are not _first_ grabbing _existing data_ and merging it into the cache. We start with some initial data, then call mergeCollection, then call it again. Before the changes we would end up skipping the initial value and merging partial data into the cache.